src/utils/FileEmitter.ts

```diff
@@ -24,6 +24,8 @@ interface OutputSpecificFileData {
 	bundle: OutputBundleWithPlaceholders;
 }
 
+const isBrowser = typeof window !== 'undefined' && typeof document !== 'undefined';
```

Comment:
I see the motivation but I am not really happy about connecting the presence of `Buffer` with the presence of some completely unrelated browser globals. A simple fix would be to just check for the presence of `Buffer`, but there is no reason this cannot work in the browser. Except for equality and type checking, we are not using any more Buffer specific features. I will add some changes and a test.